packages/create-sitecore-jss/src/templates/nextjs-sxa/src/Layout.tsx

```diff
@@ -31,24 +32,45 @@ const Layout = ({ layoutData, headLinks }: LayoutProps): JSX.Element => {
     <>
       <Scripts />
       <Head>
-        <title>{fields?.Title?.value?.toString() || 'Page'}</title>
-        <link rel="icon" href={`${publicUrl}/favicon.ico`} />
+        <title>
+        <% if (xmcloud) {-%>
+          {layoutData.sitecore.context.renderingType === 'component'
+            ? 'Sitecore Component Library'
+            : fields?.Title?.value?.toString() || 'Page'}
+          <% } else { -%>
+          {fields?.Title?.value?.toString() || 'Page'}
+        <% } -%>
+        </title>
+        <link rel="icon" href="/favicon.ico" />
         {headLinks.map((headLink) => (
           <link rel={headLink.rel} key={headLink.href} href={headLink.href} />
         ))}
       </Head>
 
       {/* root placeholder for the app, which we add components to using route data */}
       <div className={mainClassPageEditing}>
-        <header>
-          <div id="header">{route && <Placeholder name="headless-header" rendering={route} />}</div>
-        </header>
-        <main>
-          <div id="content">{route && <Placeholder name="headless-main" rendering={route} />}</div>
-        </main>
-        <footer>
-          <div id="footer">{route && <Placeholder name="headless-footer" rendering={route} />}</div>
-        </footer>
+      <% if (xmcloud) {-%>
+      {layoutData.sitecore.context.renderingType === 'component' ? (
+        <ComponentLibrary {...layoutData} />
```

Comment:
In the ticket description there was a proposal to renderer component library inline, rather than conditionally switch between component and regular layout. The assumption was, `ComponentLibrary` itself can check if it is a regular rendering and output `null`. And the some goes with regular `Placeholders` in the `Layout`, assumption that they output nothing when there is no components in them.
Did you try this approach?
 ```suggestion        
        <div id="content">
                <ComponentLibrary {...layoutData} />
                {route && <Placeholder name="headless-main" rendering={route} />}
         </div>
```

Reply:
Yes i did try this, it works but then in library mode it tries to render the `headless-header`, `headless-main` and `headless-footer` which are empty and complains about them in the logs